crates/node-builder/Cargo.toml

```diff
@@ -0,0 +1,19 @@
+[package]
+name = "reth-node-builder"
```

Comment:
having a new crate for this is useful.

since a lot of these types will be shared across the codebase we should split the crates:

perhaps:
`reth-node-(builder?)-api`: here we add all traits, similar to the interfaces crate but less bloated
and `reth-node-builder`